packages/truffle-core/lib/console.js

```diff
@@ -1,248 +1,225 @@
-  // It's here to ensure the repl manager instance gets
-  // passed down to commands.
-  this.options.repl = this.repl;
-
-  this.provision(function(err, abstractions) {
-    if (err) {
-      self.options.logger.log(
-        "Unexpected error: Cannot provision contracts while instantiating the console."
-      );
-      self.options.logger.log(err.stack || err.message || err);
-    }
-
-    self.repl.start({
-      prompt: "truffle(" + self.options.network + ")> ",
-      context: {
-        web3: self.web3
-      },
-      interpreter: self.interpret.bind(self),
-      done: callback
-    });
-
-    self.resetContractsInConsoleContext(abstractions);
-  });
-};
+  start(callback) {
+    if (!this.repl) this.repl = new Repl(this.options);
+
+    // TODO: This should probalby be elsewhere.
+    // It's here to ensure the repl manager instance gets
+    // passed down to commands.
+    this.options.repl = this.repl;
+
+    this.provision()
+      .then(abstractions => {
+        this.repl.start({
+          prompt: "truffle(" + this.options.network + ")> ",
+          context: {
+            web3: this.web3
+          },
+          interpreter: this.interpret.bind(this),
+          done: callback
+        });
 
-Console.prototype.provision = function(callback) {
-  var self = this;
+        this.resetContractsInConsoleContext(abstractions);
+      })
+      .catch(error => {
+        this.options.logger.log(
+          "Unexpected error: Cannot provision contracts while instantiating the console."
+        );
+        this.options.logger.log(error.stack || error.message || error);
+      });
+  }
 
-  fs.readdir(this.options.contracts_build_directory, function(err, files) {
-    if (err) {
+  async provision() {
+    let files;
+    try {
+      files = fse.readdirSync(this.options.contracts_build_directory);
+    } catch (error) {
       // Error reading the build directory? Must mean it doesn't exist or we don't have access to it.
       // Couldn't provision the contracts if we wanted. It's possible we're hiding very rare FS
       // errors, but that's better than showing the user error messages that will be "build folder
       // doesn't exist" 99.9% of the time.
     }
 
-    var promises = [];
+    let promises = [];
     files = files || [];
 
-    files.forEach(function(file) {
+    files.forEach(file => {
       promises.push(
-        new Promise(function(accept, reject) {
-          fs.readFile(
-            path.join(self.options.contracts_build_directory, file),
-            "utf8",
-            function(err, body) {
-              if (err) return reject(err);
-              try {
-                body = JSON.parse(body);
-              } catch (e) {
-                return reject(
-                  new Error("Cannot parse " + file + ": " + e.message)
-                );
-              }
-
-              accept(body);
-            }
-          );
-        })
+        fse
+          .readFile(
```

Comment:
can we use `fse.readFileSync` here?
that way we can keep `provision` a sync method and remove `Promise.all` (maybe rename the `promises` array to `jsonBlobs`).

Reply:
Sure! However, I wonder where it is especially useful to use async methods for performance perks...any thoughts @gnidan?